devito/ir/iet/visitors.py

```diff
@@ -152,10 +152,10 @@ def _args_decl(self, args):
         """Generate cgen declarations from an iterable of symbols and expressions."""
         ret = []
         for i in args:
-            if i.is_Tensor:
-                ret.append(c.Value('%srestrict' % i._C_typename, i._C_name))
-            elif i.is_AbstractObject or i.is_Symbol:
+            if i.is_AbstractObject or i.is_Symbol:
```

Comment:
I guess there's a rational behind this flipping...?

Reply:
Same reason `Array` has `is_Symbol` set to `True`.

I want OPS `Array` types to fall in that if clause and not the `i.is_Tensor` case

Reply:
are you just trying to avoid the `restrict` ?

Reply:
yes

Reply:
The wouldn't it be better to have `restrict` as a backend keyword rather than "cheating" in if?

Reply:
yeah, ``restrict`` should somewhat be moved to the backends (core, ops, etc) but I'm not sure it's in the scope of this PR

Reply:
This is irrelevant now with the new OPS API that uses `ACC<T>` objects instead of arrays and accessor macros. I'm working on a new PR that uses that API. This could be closed if there's no objections? @FabioLuporini

Reply:
sure, go ahead, thanks

Reply:
As we talked in the channel with Istvan about this OPS update, he said that the `OPSACC#` will continue to be supported. I don't see a reason to change it as you have done so much with it @vincepandolfo.  IMO.